### ISSUE TITLE
revts() and theme_tree2() broken when collapsing nodes

### ISSUE THREAD

**Issue by bvernot** [YuLab-SMU/ggtree#296]
I am trying to plot a tree with:

- a time scale (theme_tree2())
- reverse the scale (revts())
- a collapsed internal node (collapse())

All of the plots in this code work, up until I try to combine the three items above in a single plot. At that point, I get a nearly empty plot, with just a single line for the x-axis (but no numbers/ticks on the axis).

I used to get a warning:

```
Warning message:
Removed 1 rows containing missing values (geom_point_g_gtree). 
```

But that has gone away now - I'm not really sure why. 

Here is the code that causes the issue:

```
# ggtree version 1.16.6
library(ggplot2)
library(ggtree)
library(treeio)

## make an arbitrary tree
a = read.newick(text = '((A:1,B:1)X:1,(D:3,C:3)Y:2);')

## select nodes to collapse
node.y.idx = nodeid(a, 'Y')
node.x.idx = nodeid(a, 'X')

## works
ggtree(a)

## works [collapsing a node]
ggtree(a) %>% collapse(node.y.idx)

## works [collapsing, with scale]
p <- ggtree(a) %>% collapse(node.y.idx)
p + 
  # geom_point2(aes(subset=(node==node.y.idx)), shape=23, size=5, fill='black') +
  theme_tree2()

## works [no collapsing, with reversed scale]
p1 <- ggtree(a)
revts(p1) + 
  theme_tree2()

## does not work - no tree, and no numbers/ticks on x axis 
## [collapsing, with reversed scale]
p1 <- ggtree(a) %>% collapse(node.y.idx)
revts(p1) + 
  theme_tree2()

## does not work - no tree, and no numbers/ticks on x axis 
## [collapsing, with reversed scale, collapsing a different node]
p1 <- ggtree(a) %>% collapse(node.x.idx)
revts(p1) + 
  theme_tree2()

## Also does not show tree - but the numbers/ticks on x axis appear
## [collapsing, with reversed scale, and scale_x_continuous]
p1 <- ggtree(a) %>% collapse(node.y.idx)
revts(p1) +
  theme_tree2() +
  scale_x_continuous(limits = c(-5,-2), labels=abs)
```

**Comment by brj1** [comment]
The current version of ggtree on Bioconductor is v2.0.4. According to your comment it looks like you are using ggtree v1.16.1. Please update to the newest version before making issues.

```R
BiocManager::install("ggtree")
```

**Comment by bvernot** [comment]
Yes - I thought I had updated - I ran exactly that code before posting my issue - and it took me to 1.16.6. I see now on the bioconductor page that there's a later version, but I don't know why I'm not getting it. I will try to figure that out. So you're saying that with the most recent version these do work?

```
> BiocManager::install("ggtree")
Bioconductor version 3.9 (BiocManager 1.30.10), R 3.6.1 (2019-07-05)
Installing package(s) 'ggtree'
trying URL 'https://bioconductor.org/packages/3.9/bioc/bin/macosx/el-capitan/contrib/3.6/ggtree_1.16.6.tgz'
Content type 'application/x-gzip' length 685419 bytes (669 KB)
==================================================
downloaded 669 KB
```

Also, the last example is in the range. This works (without the collapse), although it cuts off some of the branches [the ones I was trying to collapse], which are out of range:

```
p1 <- ggtree(a)
revts(p1) +
  theme_tree2() +
  scale_x_continuous(limits = c(-5,-2), labels=abs)
```

**Comment by bvernot** [comment]
I realized I had to update my bioconductor to get to the most recent version - I've done so, and the issue is resolved. Sorry to bother you - I really thought that I had updated to the most recent version.

Also, now that collapse, revts and theme_tree2 work together, I see that collapsing a node changes the coordinate space when using revts - so you're right, the last example is not in range.

Cheers,
Ben
